documentation/src/test/java/org/hibernate/userguide/fetching/GraphParsingTest.java

```diff
@@ -0,0 +1,235 @@
+		//tag::fetching-strategies-dynamic-fetching-entity-graph-parsing-key-example-1[]
+		final EntityGraph<Movie> graph = GraphParser.parse(
+				Movie.class,
+				"cast.key( name )",
+				entityManager
+		);
+		//end::fetching-strategies-dynamic-fetching-entity-graph-parsing-key-example-1[]
+
+		Assert.assertNotNull( graph );
+	}
+
+	@Test
+	public void testEntityKeyParsing() {
+		EntityManager entityManager = getOrCreateEntityManager();
+		//tag::fetching-strategies-dynamic-fetching-entity-graph-parsing-key-example-2[]
+		final EntityGraph<Ticket> graph = GraphParser.parse(
+				Ticket.class,
+				"showing.key( movie( cast ) )",
+				entityManager
+		);
+		//end::fetching-strategies-dynamic-fetching-entity-graph-parsing-key-example-2[]
+
+		Assert.assertNotNull( graph );
+	}
+	
+	@Test
+	public void testMergingExample() {
+		EntityManager entityManager = getOrCreateEntityManager();
+		//tag::fetching-strategies-dynamic-fetching-entity-graph-merging-example[]
+		final EntityGraph<Project> a = GraphParser.parse(
+				Project.class, "employees( username )", entityManager
+		);
+	
+		final EntityGraph<Project> b = GraphParser.parse(
+				Project.class, "employees( password, accessLevel )", entityManager
+		);
+	
+		final EntityGraph<Project> c = GraphParser.parse(
+				Project.class, "employees( department( employees( username ) ) )", entityManager
+		);
+		
+		final EntityGraph<Project> all = EntityGraphs.merge( entityManager, Project.class, a, b, c );
+		//end::fetching-strategies-dynamic-fetching-entity-graph-merging-example[]
+		
+		final EntityGraph<Project> expected = GraphParser.parse(
+				Project.class,
+				"employees( username, password, accessLevel, department( employees( username ) ) )",
+				entityManager
+		);
+		
+		Assert.assertTrue( EntityGraphs.areEqual( expected, all ) );
+	}
+	
+	@Test
+	public void testFindExample() {
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			Long userId = 1L;
+
+			//tag::fetching-strategies-dynamic-fetching-entity-graph-apply-example-find[]
+			entityManager.find(
+					Employee.class,
+					userId,
+					Collections.singletonMap(
+							GraphSemantic.FETCH.getJpaHintName(),
+							GraphParser.parse( Employee.class, "username, accessLevel, department", entityManager )
+					)
+			);
+			//end::fetching-strategies-dynamic-fetching-entity-graph-apply-example-find[]
+		} );
+	}
+	
+	@Test
+	public void testQueryExample() {
+		doInJPA( this::entityManagerFactory, entityManager -> {
+			//tag::fetching-strategies-dynamic-fetching-entity-graph-apply-example-query[]
+			final String graphString = "username, accessLevel";
+			final String queryString = "select e from Employee e where e.id = 1";
+			
+			final EntityGraph<Employee> graph = GraphParser.parse(
+					Employee.class,
+					graphString,
+					entityManager
+			);
+			
+			TypedQuery<Employee> query1 = entityManager.createQuery( queryString, Employee.class );
+			query1.setHint(
```

Comment:
Did you not "migrate" the method(s) to properly set these graphs without having to know about FETCH/LOAD hints/enums? 

Ideal case, of course, would have been, of course, that what is in lines 162-166 here could be stated as simply as:

`TypedQuery<Employee> query1 = entityManager.createQuery( graph, queryString, Employee.class );`

or:

`TypedQuery<Employee> query1 = entityManager.createQuery( GraphSemantic.FETCH, graph, queryString, Employee.class );`

Submitted approach (that did not modify existing Hibernate classes) used a utility method in EntityGraphs such as:

`EntityGraphs.setFetchGraph(query1, graph);`